Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/resource/builder.go

```diff
@@ -440,15 +440,27 @@ func (b *Builder) resourceMappings() ([]*meta.RESTMapping, error) {
 	}
 	mappings := []*meta.RESTMapping{}
 	for _, r := range b.resources {
-		gvk, err := b.mapper.KindFor(unversioned.GroupVersionResource{Resource: r})
+		gvks, err := b.mapper.KindsFor(unversioned.GroupVersionResource{Resource: r})
 		if err != nil {
 			return nil, err
 		}
-		mapping, err := b.mapper.RESTMapping(gvk.GroupKind(), gvk.Version)
-		if err != nil {
-			return nil, err
+		// the list is in most-preferred to least preferred, so iterate until we find a hit
+		var firstErr error
+		for _, gvk := range gvks {
```

Comment:
> Could this be another method in the restmapper?

Maybe, but not in this pull.  It seems like someone wants a `RESTMappingForPartialResource`, but should still give back a list.